frontend/routes/notifications/notifications.component.tsx

```diff
@@ -0,0 +1,179 @@
+/**
+ *  Copyright (C) 2018 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import * as React from "react";
+import Drawer from "@material-ui/core/Drawer";
+import Typography from '@material-ui/core/Typography';
+import Icon from "@material-ui/core/Icon";
+import {INotification} from "./notification.item";
+import { Button, List, IconButton, Menu, MenuItem, Badge} from "@material-ui/core";
+import { MuiThemeProvider } from "@material-ui/core/styles";
+import { MuiTheme } from "../../styles";
+import { ListSubheaderToolbar } from "../components/listSubheaderToolbar/listSubheaderToolbar.component";
+import { NotificationEmptyItem } from "./notifications.emptyItem";
+import { NotificationsPanel } from "./notifications.panel";
+import { simpleDate, getSunday } from "../../components/utils/js/utils.filter";
+import { BarIconButton, UserActionButton, ItemLabel } from "../components/components.styles";
+import {toArray, groupBy, sortBy } from "lodash";
+import { NotificationsPanelItem, NotificationsPanelHeader } from "./notifications.panel.styles";
+
+// Props bound in <file://./notifications.container.ts>
+interface IProps {
+	sendGetNotifications: () => void ; // Bound to redux action sendGetNotifications
+	confirmSendDeleteAllNotifications: () => void ; // Bound to redux saga action confirmSendDeleteAllNotifications
+	sendUpdateNotificationRead: (id: string, read: boolean) => void; // Bound to redux saga sendUpdateNotificationRead
+	sendDeleteNotification: (id: string) => void; // Bound to redux saga sendDeleteNotification
+	notifications: INotification[]; // Bound to store state notifications
+}
+
+const NotificationsDate = () =>
+	(<NotificationsPanelItem>
+			<NotificationsPanelHeader>
+				<ItemLabel>
+					{simpleDate(new Date())}
+				</ItemLabel>
+				<ItemLabel/>
+			</NotificationsPanelHeader>
+	</NotificationsPanelItem>);
+
+export class Notifications extends React.PureComponent<IProps, any> {
+	public state = {
+		open: false,
+		menuElement: null
+	};
+
+	public componentDidMount = () => {
+		// This will download notifications from the server and save to the store on init
+		this.props.sendGetNotifications();
+	}
+
+	public toggleDrawer = (e: React.SyntheticEvent) => {
+		this.setState(Object.assign({open: !this.state.open }));
```

Comment:
Are the `event` param and Object.assign necessary?

Reply:
Nope. Will change.